src/ape/api/transactions.py

```diff
@@ -253,24 +246,24 @@ def await_confirmations(self) -> "ReceiptAPI":
             return self
 
         # Wait for nonce from provider to increment.
-        sender_nonce = self.provider.get_nonce(self.sender)
+        sender_nonce = self.provider.get_nonce(self.raw_transaction.sender)
         iterations_timeout = 20
         iteration = 0
 
-        while sender_nonce == self.nonce:  # type: ignore
+        while sender_nonce == self.raw_transaction.nonce:  # type: ignore
             time.sleep(1)
-            sender_nonce = self.provider.get_nonce(self.sender)
+            sender_nonce = self.provider.get_nonce(self.raw_transaction.sender)
             iteration += 1
             if iteration == iterations_timeout:
                 raise TransactionError(message="Timeout waiting for sender's nonce to increase.")
 
-        if self.required_confirmations == 0:
+        if self.raw_transaction.required_confirmations == 0:
```

Comment:
I'm good with this as an intermediate solution and later refactoring `required_confirmations` on a future ticket. I think ultimately we want to try to move this field to `NetworkAPI` as part of the network configuration, and then you'd have access to it through `self.network.required_confirmations`. 

Is that a correct assumption @fubuloubu ?

Reply:
Yes, it should basically be a config variable per-network e.g.:

```yaml
ethereum:
  mainnet:
    required_confirmations: 10
```

Reply:
I am marking this as resolved, if you have other questions on this comment, just mark it as unresolved and continue the conversation.